extensions/amp-twitter/1.0/component.js

```diff
@@ -27,6 +27,7 @@ function TwitterWithRef(
     timelineSourceType,
     timelineUserId,
     title,
+    togglePlaceholder,
```

Comment:
While this approach works, the concept of a placeholder is somewhat specific to AMP, and we don't need to be as prescriptive as to assume that an author wants to implement the component this way.

I'd recommend that the Preact layer have a more generic prop, such as `onLoad` and `onError`, which are called at appropriate lifecycle states -- perhaps `onLoad` once the `height` is received and `onError` when receiving a [`NO_CONTENT` message like in `0.1`](https://github.com/ampproject/amphtml/blob/24174aa4c722d020f276da6e0bf5f64f0e788b73/extensions/amp-twitter/0.1/amp-twitter.js#L100). And then the AMP layer would pass the placeholder accordingly:
```js
// amp-twitter.js
onError: () => this.togglePlaceholder(true),
onLoad: () => this.togglePlaceholder(false)
```

Reply:
i like this approach better - updated to use do that